components/search/BookingInfoSearch/BookingInfoSearchUIComplete.tsx

```diff
@@ -1,57 +1,97 @@
 "use client";
+import { useState } from "react";
 import AdultsNumberInput from "./SearchComponents/AdultsNumberInput";
 import { DatePickerWithRange } from "./DatePickerWithRange";
-
-type BookingInfo = {
-  checkinDate: string;
-  checkoutDate: string;
-  adultsNumber: number;
-  numDays: number;
-};
+import { searchParamsType } from "@/app/page";
 
 type BookingInfoUISearchCompleteProps = {
-  bookingInfo: BookingInfo;
-  setBookingInfo: (info: BookingInfo) => void;
+  bookingInfo: searchParamsType;
+  setBookingInfo: (info: searchParamsType) => void;
 };
 
-const BookingInfoUISearchComplete: React.FC<BookingInfoUISearchCompleteProps> = ({
-  bookingInfo,
-  setBookingInfo,
-}) => {
+const BookingInfoUISearchComplete: React.FC<
+  BookingInfoUISearchCompleteProps
+> = ({ bookingInfo, setBookingInfo }) => {
+  const [tempBookingInfo, setTempBookingInfo] = useState(bookingInfo);
+  const [isDateValid, setIsDateValid] = useState(true);
```

Comment:
I think we can assume a date is always not fully set or valid. This responsibility should fall onto the date range picker component.